### DISCUSSION TITLE
wezterm ssh in-line?

### DISCUSSION THREAD

**Discussion by panki27** [wezterm/wezterm#5535] [Q&A]
### What Operating System(s) are you running on?

Windows, Linux X11

### Which Wayland compositor or X11 Window manager(s) are you using?

the default that comes with XFCE on debian... no idea tbh.

### WezTerm version

20240203-110809-5046fc22

### Ask your question!

`wezterm` has completely replaced `tmux` for me. What I always wanted was a "ssh aware" tmux that would let me create remote splits on the fly - and `wezterm ssh` *does* let me do that.

However, `wezterm ssh` always seems to spawn a new window. Is it possible to not do this, but instead have the `wezterm ssh` session spawn inside the pane I've run the command in?

I have not found anything in the docs indicating this is possible - if it isn't, what's hindering it?

**Comment by wez** [comment]
If you're using https://wezfurlong.org/wezterm/config/lua/wezterm/default_ssh_domains.html then you can do `wezterm cli spawn --domain-name 'SSH:hostname'` to connect to a host from your ssh config.

See also:
* https://wezfurlong.org/wezterm/cli/cli/spawn.html
* https://wezfurlong.org/wezterm/cli/cli/split-pane.html

**Reply by DivineMK** [reply]
I have similar question and tried your suggestion, but it is still not what I wanted. Using `wezterm cli spawn --domain-name` will spawn it in a new tab, but I would like to be able to spawn in the same tab, either in same pane or different panes. How can I do this?

**Reply by wez** [reply]
It's a bit clunky, but after you have spawned that, you can use `wezterm cli split-pane --pane-id TO_SPLIT --move-pane-id SSH_PANE` to move that newly spawned pane into the place you want it.  See `wezterm cli split-pane --help`

**Reply by DivineMK** [reply]
I see. I wonder if there is a way to do this within one command? Otherwise is it possible to spawn ssh in the current pane, similar to `kitty kitten ssh`?

**Reply by eraserix** [reply]
I put together the following script to have to ssh aware niceties of `wezterm ssh` in the current pane.

I requires the host to be in .ssh/config (the example uses reptile as host entry.

```sh
#!/bin/sh

np=$(wezterm cli spawn --domain-name SSH:reptile)

wezterm cli split-pane --move-pane-id $np
wezterm cli kill-pane
```
